### ISSUE TITLE
end of buffer exception

### ISSUE THREAD

**Issue by suhuangjian** [mapbox/mapbox-gl-native#13477]
### My code
```Kotlin
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        Mapbox.getInstance(this, "My key")
        // Create supportMapFragment
        val mapFragment: SupportMapFragment
        if (savedInstanceState == null) {

            // Create fragment
            val transaction = supportFragmentManager.beginTransaction()

            val patagonia = LatLng(-52.6885, -70.1395)

            // Build mapboxMap
            val options = MapboxMapOptions()
            options.styleUrl(Style.SATELLITE)
            options.camera(
                CameraPosition.Builder()
                    .target(patagonia)
                    .zoom(9.0)
                    .build()
            )

            // Create map fragment
            mapFragment = SupportMapFragment.newInstance(options)

            // Add map fragment to parent container
            transaction.add(R.id.container, mapFragment, "com.mapbox.map")
            transaction.commit()
        } else {
            mapFragment = supportFragmentManager.findFragmentByTag("com.mapbox.map") as SupportMapFragment
        }
    }
}
```
### Steps to reproduce

 1. Run the  simple demo
 2. Power cut when the logo shows but map not yet show
 3. Restart the app
 4. App crash each time
### Actual behavior

    java.lang.Error: end of buffer exception
        at com.mapbox.mapboxsdk.maps.renderer.MapRenderer.nativeRender(Native Method)
        at com.mapbox.mapboxsdk.maps.renderer.MapRenderer.onDrawFrame(MapRenderer.java:74)
        at com.mapbox.mapboxsdk.maps.renderer.glsurfaceview.GLSurfaceViewMapRenderer.onDrawFrame(GLSurfaceViewMapRenderer.java:58)
        at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1548)
        at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1259)

### Configuration

**Android versions:**  Android 7.1.2
**Device models:**  RK3399
**Mapbox SDK versions:**  6.7.1

**Comment by tobrun** [comment]
@suhuangjian thank you for reaching out and reporting your problem.
Could you clarify what you mean with `Power cut`? 
Would love to try reproducing this issue.

**Comment by suhuangjian** [comment]
Sorry，it is my fault.
Because our Android device is a vehicle terminal.
So our terminal is no battery, the device is directly connected to the vehicle AC power.
The power cut means Unplug the power cable and the Android device will shut downimmediately witout normal shutdown

**Comment by tobrun** [comment]
This is a race condition with our binary shaders. Binary shaders have been disabled in https://github.com/mapbox/mapbox-gl-native/pull/14298 so it won't crash anymore. Follow up on enabling them in long term future is ticketed in https://github.com/mapbox/mapbox-gl-native/issues/14294.

This fix is available in the `v7.4.0` release and is currently being backported to v6.8.x in https://github.com/mapbox/mapbox-gl-native/pull/14672 (which we aim to release somewhere this week).
